enhancements/network/persistent-ips-sdn-secondary-networks.md

```diff
@@ -0,0 +1,1006 @@
+  allow for persistent IPs
+
+#### Stopping a Virtual Machine
+```mermaid
+sequenceDiagram
+actor user
+
+user ->> KubeVirt: stopVM(vm=vmA)
+
+KubeVirt ->> Kubernetes: deletePod(owner=vmA)
+Kubernetes -->> KubeVirt: OK
+
+KubeVirt -->> user: OK
+
+note over OVN-K: notices a pod was deleted
+OVN-K ->> OVN-K: ips = teardownOVNEntities()
+
+alt is deleted pod orphaned ?
+OVN-K ->> Kubernetes:   deletePersistentIPs(ips)
+Kubernetes -->> OVN-K: OK
+end
+```
+
+The main difference from a traditional IPAM CNI response to a pod delete is the
+`IPAMLease` allocation will not be deleted if the CNI DEL arguments
+feature any owner indication. Thus, on the VM scenario, the `IPAMLease`s
+will **not** be deleted.
+
+#### Starting a (previously stopped) Virtual Machine
+```mermaid
+sequenceDiagram
+
+actor user
+
+user ->> KubeVirt: startVM(vm=vmA)
+
+KubeVirt ->> Kubernetes: createPod(name=<generated name>, owner=vmA)
+
+
+note over OVN-K: notices a new pod was added having an owner
+OVN-K ->> Kubernetes: getPersistentIPs(owner=vmA)
+Kubernetes -->> OVN-K: persistentIPs
+OVN-K ->> Kubernetes: result(..., ips = persistentIPs)
+
+Kubernetes -->> KubeVirt: OK
+
+KubeVirt -->> user: OK
+```
+
+When creating the pod, the OVN-Kubernetes IPAM module finds existing
+`IPAMLease`s for the workload. It will thus use those already reserved
+allocations, instead of generating brand new allocations for the pod where the
+encapsulating object will run.
+
+#### Removing a Virtual Machine
+```mermaid
+sequenceDiagram
+
+actor user
+
+user ->> KubeVirt: stopVM(vm=vmA)
+
+KubeVirt ->> Kubernetes: deletePod(owner=vmA)
+
+note over OVN-K: notices a pod was deleted
+OVN-K ->> OVN-K: ips = teardownOVNEntities()
+
+Kubernetes -->> KubeVirt: OK
+
+KubeVirt -->> user: OK
+
+loop persistentIPs with owner=vmA:
+    Kubernetes ->> Kubernetes: garbageCollect(allocation)
+end
+```
+
+This scenario, from the OVN-Kubernetes IPAM module perspective is exactly like
+deleting a pod: since the `IPAMLease` is owned by a Kubernetes entity we
+give up on manipulating the datastore for it, and delegate to Kubernetes to
+garbage collect them once the Virtual Machine is removed.
+
+### API Extensions
+
+- Adds a new `IPAMLease` CRD. OVN-Kubernetes will provision these to
+  "block" the IP allocation during the VM object lifecycle.
+  - It must be "owned" by the pod's encapsulating object (i.e. VM) - thus
+  allowing Kubernetes to GC it once they're gone. To comply with Kubernetes GC,
+  the `IPAMLease` **must** be created in the same namespace as its owner.
+  - It must be "retrievable" by OVN-Kubernetes (i.e. so the "old" VM gets the
+  same IP when in runs in a "new" pod). For that, we will compose a unique key
+  with the required information, allowing the clients to get the CR by name.
+
+This is the proposed spec of the CRD:
+```yaml
+apiVersion: "persistentip.cni.cncf.io/v1alpha1"
+kind: IPAMLease
+metadata:
+  name: vm-a.tenantred.pod16367aacb67
```

Comment:
yeah, it is enough. It does depend on who's creating the object ... Like, if it is OVN-K creating it, it cannot use the vm spec interface name, because it doesn't know the vmi spec. Right ? 

Where would the CNI plugin get that information from ? ... Passing the key name over CNI args is doable FWIW.

Reply:
In case ovn-k create IPAMClaim we need to pass the IPAMClaim key too since ovn-k do not know the VM interface name.

So in the part of this document where ovn-k create it's we have to pass the key.